thrust/iterator/transform_input_output_iterator.h

```diff
@@ -85,23 +85,25 @@ THRUST_NAMESPACE_BEGIN
  */
 
 template <typename InputFunction, typename OutputFunction, typename Iterator>
-  class transform_input_output_iterator
+class transform_input_output_iterator
     : public detail::transform_input_output_iterator_base<InputFunction, OutputFunction, Iterator>::type
 {
 
   /*! \cond
    */
 
-  public:
-
-    typedef typename
-    detail::transform_input_output_iterator_base<InputFunction, OutputFunction, Iterator>::type
-    super_t;
+public:
+  typedef typename detail::
+    transform_input_output_iterator_base<InputFunction, OutputFunction, Iterator>::type super_t;
 
-    friend class thrust::iterator_core_access;
+  friend class thrust::iterator_core_access;
   /*! \endcond
    */
 
+  /*! Null constructor does nothing.
+   */
+  __host__ __device__ transform_input_output_iterator() {}
```

Comment:
Rather than defining an empty constructor we should default it

```suggestion
 transform_input_output_iterator() = default
```

Reply:
Hmmm, I thought I had tried that locally and it failed (something to do with `__host__ __device__`). But I tried now and it works, so I must have misremembered.

Reply:
Also I was copying the style (including verbatim comment) of the other Thrust iterator classes.

Reply:
> Hmmm, I thought I had tried that locally and it failed (something to do with `__host__ __device__`). But I tried now and it works, so I must have misremembered.

nvcc does not compile `__host__ __device__ meow() = default` but only `meow() = default;` Maybe that was the issue?